proof/file.go

```diff
@@ -36,6 +40,27 @@ type Version uint32
 const (
 	// V0 is the first version of the proof file.
 	V0 Version = 0
+
+	// FileMaxNumProofs is the maximum number of proofs we expect/allow to
+	// be encoded within a single proof file. Given that there can only be
+	// one transfer per block, this value would be enough to transfer an
+	// asset every 10 minutes for 8 years straight. This limitation might be
+	// lifted at some point when proofs can be compressed into a single
+	// zero-knowledge proof.
+	FileMaxNumProofs = 420000
+
+	// FileMaxProofSizeBytes is the maximum size of a single proof in a
+	// proof file. The maximum size of a meta reveal is 1 MiB, so this value
+	// would cap the number of additional inputs within a proof to roughly
+	// 16 of assets with such large meta data.
+	FileMaxProofSizeBytes = 16 * MetaDataMaxSizeBytes
```

Comment:
Not sure about this one tbh. Since proof size is meta reveal + size of each transition proof. Even with a small meta reveal, if an asset was part of a 'wide' TX like an exchange batched withdraw or coinjoin then its proof file would be a fair bit bigger. IIUC this limit would apply to a single transition proof including proofs for `AdditionalInputs`, but wondering if it could be applied to each `AdditionalInput` indvidually.

Not urgent / can be bumped later client-side.

Reply:
Yeah, it's quite hard to pick good numbers... I bumped this to `128 MiB` for now. But still, worst case that means we allocate 128 mebibytes for every attacker that uploads a malicious proof file to an RPC. Currently there is no such RPC that isn't protected by macaroon or by dev mode (the universe RPCs all only accept a single proof, not a file). So we should be okay, but still...

Reply:
Ah, yeah I would think that maybe the RPC limit is lower than this value, which is global IIUC. E.x. I think this limit would prevent creating transfers with large resulting proofs, even though transfers are macaroon-protected.

Reply:
You mean the general RPC message size? We use 200MB as we do in `lnd`: https://github.com/lightninglabs/taproot-assets/blob/a488519854bb278db82de686629d4fdd2abd226f/server.go#L263